src/app_engine/decline_page.py

```diff
@@ -12,36 +12,54 @@
 import webapp2
 
 import constants
+import gcm_notify
 import gcmrecord
 import room
 import util
 
+
 class DeclinePage(webapp2.RequestHandler):
   def write_response(self, result):
     self.response.write(json.dumps({
-      'result': result
+        'result': result
     }))
-  
+
   def post(self, room_id):
     msg = util.get_message_from_json(self.request.body)
     if not util.has_msg_field(msg, constants.PARAM_CALLEE_GCM_ID, basestring):
       self.write_response(constants.RESPONSE_INVALID_ARGUMENT)
       return
-
     callee_gcm_id = msg[constants.PARAM_CALLEE_GCM_ID]
 
-    # Look up and validate callee by gcm id.
-    # TODO (chuckhays): Once registration is enabled, turn on restriction
+    # Remove the room.
+    result, declined_room = room.remove_room_for_declined_call(
+        self.request.host_url, room_id, callee_gcm_id)
+    if result != constants.RESPONSE_SUCCESS:
+      self.write_response(result)
+      return
+
+    # We successfully removed the room from memcache, but should still have a
+    # copy of the removed room. Use that to get the caller's GCM id.
+    caller_gcm_id = declined_room.get_other_client_id(callee_gcm_id)
+
+    # Retrieve callee's other GCM ids.
+    # TODO(chuckhays): Once registration is enabled, turn on restriction
     # to only return verified records.
-    callee_records = gcmrecord.GCMRecord.get_by_gcm_id(callee_gcm_id, False)
+    callee_id = gcmrecord.GCMRecord.get_user_id_for_gcm_id(callee_gcm_id, False)
```

Comment:
nit: put get_user_id_for_gcm_id and get_by_user_id in a helper method of gcmrecord. We should need it in other places as well.